### PR TITLE
feat: [64] Update Spending Over Time chart with net amounts and account tooltips

### LINKED ISSUE #64: Update Spending Over Time chart: net amounts + account tooltips
## Description

Update the Spending Over Time dashboard chart to show net spending (debits and credits combined) instead of only debit transactions, and display account names in the chart tooltip when hovering over data points.

## Tasks

- [ ] Remove `direction = 'debit'` filter from the `timeSeriesData()` query
- [ ] Use `SUM(transactions.amount)` for net calculation (Basiq amounts are already signed)
- [ ] Join `accounts` table and group by account to get per-account breakdown
- [ ] Enrich data shape: add `accounts` array to each period entry with `{name, total}`
- [ ] Replace default ApexCharts tooltip with custom formatter showing account breakdown and net total
- [ ] Style tooltip for dark/light theme consistency
- [ ] Update all tests to use negative debit amounts (matching real Basiq data)
- [ ] Add tests: net calculation, positive net, zero net, account names in data

## Acceptance Criteria

- Chart reflects both debits and credits as net spending per period
- Chart line crosses above $0 when credits exceed debits in a period
- Hovering over any data point shows each account's contribution and the net total
- Periods with no transactions show empty accounts array
- All tests pass with realistic signed amounts
- Full CI green (Pint, PHPStan, Pest)

### PR DESCRIPTION
**Pull request by robwilde**
## Summary

Closes #64

- Refactored the Spending Over Time chart to show **net spending** (debits + credits) instead of only debits, using `SUM(transactions.amount)` since Basiq stores signed amounts
- Added a **custom tooltip** that shows per-account breakdown with account names, individual amounts, and net total on hover
- Joined the `accounts` table in the query and enriched the data shape to include an `accounts` array per period

## Changes

- `app/Livewire/SpendingOverTime.php` — Removed debit-only filter, joined accounts table, grouped by account for per-period breakdown, extracted `fillPeriods()` helper
- `resources/views/livewire/spending-over-time.blade.php` — Custom ApexCharts tooltip with account breakdown, dark/light theme support
- `tests/Feature/Livewire/SpendingOverTimeTest.php` — Updated all tests to use realistic negative debit amounts, added 4 new tests (net calculation, positive net, zero net, account names)

## Test plan

- [x] `op ci` passes (410 tests, PHPStan clean, Pint clean)
- [x] Visual verification via Playwright: chart y-axis shows values above and below $0
- [x] Tooltip shows account names with signed amounts and net total
- [ ] Manual review of chart on dashboard

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: resources/views/livewire/spending-over-time.blade.php [hunk 1]
```diff
@@ -75,9 +86,30 @@
...
+                                        const bc = isDark ? '#3f3f46' : '#e5e7eb'
+                                        let html = '<div style=\'' + bg + 'border:1px solid ' + bc + ';border-radius:8px;padding:10px 12px;font-size:13px;min-width:180px;\'>'
+                                        html += '<div style=\'font-weight:600;margin-bottom:6px;\'>' + self.tooltipDate(point.date, self.aggregation) + '</div>'
+
+                                        if (point.accounts && point.accounts.length > 0) {
+                                            point.accounts.forEach(function(acc) {
+                                                html += '<div style=\'display:flex;justify-content:space-between;gap:16px;padding:1px 0;\'>'
+                                                html += '<span style=\'opacity:0.7;\'>' + acc.name + '</span>'
+                                                html += '<span style=\'font-weight:500;font-variant-numeric:tabular-nums;\'>' + self.formatMoney(acc.total) + '</span>'
+                                                html += '</div>'
```
[the remark this answers is not in the document]
Comment (robwilde) [reply resources/views/livewire/spending-over-time.blade.php:108]: Fixed — added escapeHtml() helper.

File: app/Livewire/SpendingOverTime.php [hunk 2]
```diff
@@ -31,56 +32,35 @@ public function timeSeriesData(): array
...
+
+        /** @var Collection<int, stdClass> $rows */
         $rows = Transaction::query()
-            ->where('user_id', auth()->id())
-            ->where('direction', TransactionDirection::Debit)
-            ->where('post_date', '>=', $start)
-            ->selectRaw("{$groupExpression} as period_date, SUM(amount) as total")
-            ->groupBy('period_date')
+            ->join('accounts', 'transactions.account_id', '=', 'accounts.id')
+            ->where('transactions.user_id', auth()->id())
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Livewire/SpendingOverTime.php:49]: Fixed — added `whereColumn('accounts.user_id', 'transactions.user_id')` as a defensive safeguard.

File: app/Livewire/SpendingOverTime.php [hunk 3]
```diff
@@ -31,56 +32,35 @@ public function timeSeriesData(): array
         $isSqlite = $connection->getDriverName() === 'sqlite';
 
         $groupExpression = match ($aggregation) {
-            'day' => 'DATE(post_date)',
+            'day' => 'DATE(transactions.post_date)',
```
[the remark this answers is not in the document]
Comment (robwilde) [reply app/Livewire/SpendingOverTime.php:35]: Not applied — SQLite (used in tests) stores date columns with a time component (e.g. 2025-03-19 00:00:00). Removing DATE() causes the GROUP BY key to not match the Y-m-d keys in fillPeriods(), breaking 8 tests. The wrapper is necessary for cross-DB compatibility.
